client/src/pages/Checkout/Checkout.tsx

```diff
@@ -0,0 +1,127 @@
+import { useState } from 'react';
+import { Box, FormControl, OutlinedInput, Button, MenuItem, Select } from '@material-ui/core';
+
+import Navigation from '../Navigation/Navigation';
+import { useSnackBar } from '../../context/useSnackbarContext';
+import useStyles from './useStyles';
+
+interface CardInfo {
+  cardNumber: string;
+  expirationMonth: string;
+  expirationYear: string;
+  cvc: string;
+}
+
+const Checkout = (): JSX.Element => {
+  const classes = useStyles();
+  const [cardInfo, setCardInfo] = useState<CardInfo>({
+    cardNumber: '',
+    expirationMonth: '',
+    expirationYear: '',
+    cvc: '',
+  });
+
+  const months = ['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12'];
+  const years = ['21', '22', '23', '24', '25', '26', '27', '28', '29', '30'];
+
+  const { updateSnackBarMessage } = useSnackBar();
+
+  const handleChange = (e: { target: HTMLInputElement | HTMLTextAreaElement }) => {
+    const { name, value } = e.target;
+    setCardInfo({ ...cardInfo, [name]: value });
+  };
+
+  // Had to duplicate these because TypeScript and Material-UI conflicts
+  const handleChangeMonth = async (e: React.ChangeEvent<{ value: unknown }>) => {
+    setCardInfo({ ...cardInfo, expirationMonth: e.target.value as string });
+  };
+
+  const handleChangeYear = async (e: React.ChangeEvent<{ value: unknown }>) => {
+    setCardInfo({ ...cardInfo, expirationYear: e.target.value as string });
+  };
+
+  const handleClick = (): void => {
+    const { cardNumber, expirationMonth, expirationYear, cvc } = cardInfo;
+
+    if (
+      cardNumber.trim() === '' ||
+      expirationMonth.trim() === '' ||
+      expirationYear.trim() === '' ||
+      cvc.trim() === ''
+    ) {
+      return updateSnackBarMessage('Please enter a valid card');
+    }
+  };
+
+  const renderMonths = months.map((month) => (
+    <MenuItem key={month} value={month}>
+      {month}
+    </MenuItem>
+  ));
+
+  const renderYears = years.map((year) => (
+    <MenuItem key={year} value={year}>
+      {year}
+    </MenuItem>
+  ));
+
+  return (
+    <Box className={classes.page}>
+      <Navigation />
+      <Box my={15} px={40} py={10} className={classes.checkoutBox}>
+        <Box className={classes.headerBox}>
+          <h1>Checkout</h1>
+        </Box>
+        <Box mb={2.5} className={classes.cardNumberBox}>
+          <h3>Card number</h3>
+          <FormControl>
+            <OutlinedInput
+              placeholder="7364 8257 2746 6153"
+              name="cardNumber"
+              value={cardInfo.cardNumber}
```

Comment:
Should use Formik here, it's available in the project and can really make form work more maintainable and reliable.

Reply:
Also, Yup for our validation work

Reply:
Hey Ethan, just converted the component to use formik and yup!